--- gnome-code-assistance-3.16.0/configure.ac~	2015-04-29 21:33:03.000000000 +0200
+++ gnome-code-assistance-3.16.0/configure.ac	2016-04-16 20:53:17.011283001 +0200
@@ -410,7 +410,7 @@
               [enable_vala=$enableval],
               [enable_vala=auto])
 
-BACKEND_VALA_LIBVALA_VERSIONS="libvala-0.30 libvala-0.28 libvala-0.26 libvala-0.24 libvala-0.22 libvala-0.20"
+BACKEND_VALA_LIBVALA_VERSIONS="libvala-0.32 libvala-0.30 libvala-0.28 libvala-0.26 libvala-0.24 libvala-0.22 libvala-0.20"
 BACKEND_VALA_MODULES="gobject-2.0 >= 2.36 glib-2.0 >= 2.36 gio-2.0 >= 2.36 gee-0.8 gio-unix-2.0 >= 2.36"
 
 BACKEND_VALA_LIBVALA=
